In [1]:
import numpy as np

#### Задание 1.
##### Провести дисперсионный анализ для определения того, есть ли различия среднего роста среди взрослых футболистов, хоккеистов и штангистов. Даны значения роста в трех группах случайно выбранных спортсменов: Футболисты: 173, 175, 180, 178, 177, 185, 183, 182. Хоккеисты: 177, 179, 180, 188, 177, 172, 171, 184, 180. Штангисты: 172, 173, 169, 177, 166, 180, 178, 177, 172, 166, 170.

In [2]:
gf = np.array([173, 175, 180, 178, 177, 185, 183, 182])
gh = np.array([177, 179, 180, 188, 177, 172, 171, 184, 180])
gw = np.array([172, 173, 169, 177, 166, 180, 178, 177, 172, 166, 170])

Количество надлюдений в каждой группе

In [3]:
nf = len(gf)
nh = len(gh)
nw = len(gw)
nf, nh, nw

(8, 9, 11)

Всего надлюдений в k группах

In [4]:
n_all = nf + nh + nw
k = 3
n_all, k

(28, 3)

Проведем однофакторный дисперсионный анализ. Сначала найдем средний рост в каждой группе:

In [5]:
gf_mean = np.mean(gf)
gh_mean = np.mean(gh)
gw_mean = np.mean(gw)
gf_mean, gh_mean, gw_mean

(179.125, 178.66666666666666, 172.72727272727272)

Видно, что средний рост в группах отличается. Установим, что это отличие статистически значимо. Для этого сначала соберем все значения среднего роста в один массив:

In [6]:
g_all = np.concatenate([gf, gh, gw])
g_all

array([173, 175, 180, 178, 177, 185, 183, 182, 177, 179, 180, 188, 177,
       172, 171, 184, 180, 172, 173, 169, 177, 166, 180, 178, 177, 172,
       166, 170])

Найдем средний рост по всем значениям:

In [7]:
g_all_mean = np.mean(g_all)
g_all_mean

176.46428571428572

Найдем $S^2$ — сумму квадратов отклонений наблюдений от общего среднего:

In [8]:
s2_all = np.sum((g_all - g_all_mean)**2)
s2_all

830.9642857142857

Найдем $S_{F}^{2}$ - сумму квадратов отклонений средних групповых значений от общего среднего:

In [9]:
s2_f = ((gf_mean - g_all_mean)**2) * nf + ((gh_mean - g_all_mean)**2) * nh + ((gw_mean - g_all_mean)**2) * nw
s2_f

253.9074675324678

Найдем $S_{ost}^{2}$ — остаточную сумму квадратов отклонений:

In [10]:
s2_ost = np.sum((gf - gf_mean)**2) + np.sum((gh - gh_mean)**2) + np.sum((gw - gw_mean)**2)
s2_ost

577.0568181818182

Удостоверимся, что соблюдается равенство $S^2=S_{F}^{2}+S_{ost}^{2}$:

In [11]:
s2_all, s2_f + s2_ost

(830.9642857142857, 830.964285714286)

Найдем общую дисперсию:

In [12]:
sigma2_general = s2_all / (n_all - 1)
sigma2_general

30.776455026455025

Найдем факторную дисперсию:

In [13]:
sigma2_f = s2_f / (k - 1)
sigma2_f

126.9537337662339

Найдем остаточную дисперсию:

In [14]:
sigma2_ost = s2_ost / (n_all - k)
sigma2_ost

23.08227272727273

Вычислим $F_{H}$:

In [15]:
F_h = sigma2_f / sigma2_ost
F_h

5.500053450812598

Найдем значение $F_{critical}$ в таблице критических точек распределения Фишера-Снедекора для заданного уровня значимости α=0.05 и двух степеней свободы:

$df_{between}$ = k-1 = 3-1 = 2  
$df_{internal}$ = n_all-k = 28-3 = 25

Для данных значений $F_{critical}$=3.38. Так как $F_{H}$>$F_{critical}$ (5.50>3.38), различие среднего роста в трех группах статистически значимо.

Также вычислим эмпирическое корреляционное отношение $\eta^2$:

In [16]:
eta2 = s2_f / s2_all
eta2

0.30555761769497997

Принято считать, что при значениях $\eta^2$ ниже 0.2-0.3 групповые значения средних не имеют статистически достоверного отличия. Здесь немного выше. Отличия будут или нет?

##### Средний рост штангистов намного ниже среднего роста футболистов и хоккеистов. Попробуем провести расчет без штангистов

In [17]:
g_fh = np.concatenate([gf, gh])
g_fh

array([173, 175, 180, 178, 177, 185, 183, 182, 177, 179, 180, 188, 177,
       172, 171, 184, 180])

In [18]:
g_fh_mean = np.mean(g_fh)
g_fh_mean

178.88235294117646

In [19]:
s2_fh = np.sum((g_fh - g_fh_mean)**2)
s2_fh

347.7647058823529

In [20]:
s2_f_fh = ((gf_mean - g_fh_mean)**2) * nf + ((gh_mean - g_fh_mean)**2) * nh
s2_f_fh

0.889705882352978

In [21]:
s2_ost_fh = np.sum((gf - gf_mean)**2) + np.sum((gh - gh_mean)**2)
s2_ost_fh

346.875

Удостоверимся, что соблюдается равенство $S^2=S_{F}^{2}+S_{ost}^{2}$:

In [22]:
s2_fh, s2_f_fh + s2_ost_fh

(347.7647058823529, 347.764705882353)

In [23]:
sigma2_gen_fh = s2_fh / (len(g_fh) - 1)
sigma2_gen_fh

21.735294117647058

In [24]:
sigma2_f_fh = s2_f_fh / ((k - 1) - 1)
sigma2_f_fh

0.889705882352978

In [25]:
sigma2_ost_fh = s2_ost_fh / (len(g_fh) - (k -1))
sigma2_ost_fh

23.125

In [26]:
F_h_fh = sigma2_f_fh / sigma2_ost_fh
F_h_fh

0.03847376788553418

Найдем значение $F_{critical}$ в таблице критических точек распределения Фишера-Снедекора для заданного уровня значимости α=0.05 и двух степеней свободы:

$df_{between}$ = 2-1 = 1  
$df_{internal}$ = 17-2 = 15

Для данных значений $F_{critical}$=4,45 . Так как $F_{H}$<$F_{critical}$ (0.038<4,45), различие среднего роста футболистов и хоккеистов статистически неразлечимо.

In [27]:
eta2_fh = s2_f_fh / s2_fh
eta2_fh

0.0025583558863329883

Значение $\eta^2$ близко к 0 - групповые значения среднего роста футболистов и хоккеистов не имеют статистически достоверного отличия.